src/ts/downloaders/BulkDownloader.ts

```diff
@@ -65,12 +66,22 @@ export class BulkDownloader extends Downloader {
         const postLinks: Set<string> = await this.collectImageLinks(downloadSpeed);
 
         // Collect the media files of the posts
-        const mediaLinks: string[] = await this.collectMedia(postLinks);
+        const mediaInfo: ContentResponse[] = await this.collectMedia(postLinks);
+
+        const mediaLinks: string[] = [];
+        const timestamps: number[] = [];
+        for (const media of mediaInfo) {
+            for (const url of media.mediaURL) {
+                mediaLinks.push(url);
```

Comment:
This is more a genera not. But I would change the type of `imageURL` to

```typescript
interface DownloadURL {
	url: string;
    timestamp: number
}
```

and remove the timestamp list. This way it is easier to associate the timestamp to the url.

In this case the type of the `DownloadMessage` would change to 

```typescript
export interface DownloadMessage {
    imageURL: DownloadURL[];
    accountName: string;
    timestamp: number[];
    type: DownloadType;
}
```